numba/np/arraymath.py

```diff
@@ -1595,7 +1595,11 @@ def median_impl(a):
         # workspace anyway
         temp_arry = a.flatten()
         n = temp_arry.shape[0]
-        return _median_inner(temp_arry, n)
+        if n != 0: 
+            return _median_inner(temp_arry, n)
+        else:
+            raise ValueError('Cannot compute median of an empty array')
```

Comment:
I think we need follow NumPy behavior over here and return `np.nan` instead of raising a `ValueError`
```suggestion
            return np.nan
```

(`np.median(np.array([]))` outputs `np.nan` in NumPy)